resource/matching.go

```diff
@@ -0,0 +1,107 @@
+package resource
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/aelsabbahy/goss/system"
+	"github.com/aelsabbahy/goss/util"
+)
+
+type Matching struct {
+	Title   string      `json:"title,omitempty" yaml:"title,omitempty"`
+	Meta    meta        `json:"meta,omitempty" yaml:"meta,omitempty"`
+	Content interface{} `json:"content,omitempty" yaml:"content,omitempty"`
+	Id      string      `json:"-" yaml:"-"`
+	Matches matcher     `json:"matches" yaml:"matches"`
+}
+
+type MatchingMap map[string]*Matching
+
+func (a *Matching) ID() string      { return a.Id }
+func (a *Matching) SetID(id string) { a.Id = id }
+
+// FIXME: Can this be refactored?
+func (r *Matching) GetTitle() string { return r.Title }
+func (r *Matching) GetMeta() meta    { return r.Meta }
+
+func (a *Matching) Validate(sys system.System) []TestResult {
+	skip := false
+
+	// ValidateValue expects a function
+	stub := func() (interface{}, error) {
+		return a.Content, nil
+	}
+
+	var results []TestResult
+	results = append(results, ValidateValue(a, "matches", a.Matches, stub, skip))
+	return results
+}
+
+func (ret *MatchingMap) UnmarshalJSON(data []byte) error {
```

Comment:
We can leave it as is for now. If a second non-system resource comes up, then the dummy approach will be preferable.